src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.SkipValue.cs

```diff
@@ -2,28 +2,50 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
+using System.Diagnostics;
+
 namespace System.Security.Cryptography.Encoding.Tests.Cbor
 {
     internal partial class CborReader
     {
-        public void SkipValue()
+        // flag used to temporarily disable conformance level checks,
+        // e.g. during a skip operation over nonconforming encodings.
+        private bool _isConformanceLevelCheckEnabled = true;
+
+        public void SkipValue(bool validateConformance = false) => SkipToAncestor(0, validateConformance);
+        public void SkipToParent(bool validateConformance = false)
         {
-            CborReaderCheckpoint checkpoint = CreateCheckpoint();
+            if ((_nestedDataItems?.Count ?? 0) == 0)
+            {
+                throw new InvalidOperationException("CBOR reader is at the root context.");
+            }
+
+            SkipToAncestor(1, validateConformance);
+        }
+
+        private void SkipToAncestor(int depth, bool validateConformance)
+        {
+            Debug.Assert(0 <= depth && depth <= (_nestedDataItems?.Count ?? 0));
+            Checkpoint checkpoint = CreateCheckpoint();
+            _isConformanceLevelCheckEnabled = validateConformance;
 
             try
             {
-                int depth = 0;
-
                 do
                 {
                     SkipNextNode(ref depth);
                 } while (depth > 0);
             }
             catch
             {
-                RestoreCheckpoint(checkpoint);
+                RestoreCheckpoint(in checkpoint);
                 throw;
             }
+            finally
+            {
+                _isConformanceLevelCheckEnabled = false;
```

Comment:
Are there any tests that check for conformance after a call to skip?  Because this feels backwards to me.

Reply:
Clearly there aren't any yet :D